calliope/backend/backends.py

```diff
@@ -654,7 +654,11 @@ def get_parameter(
     ) -> Optional[xr.DataArray]:
         parameter = self.parameters.get(parameter_name, None)
         if isinstance(parameter, xr.DataArray) and not as_backend_objs:
-            return self.apply_func(self._from_pyomo_param, parameter)
+            param_as_vals = self.apply_func(self._from_pyomo_param, parameter)
+            if parameter.original_dtype.kind == "M":
+                return xr.apply_ufunc(pd.to_datetime, param_as_vals)
+            else:
+                return param_as_vals.astype(parameter.original_dtype)
         else:
             return parameter
```

Comment:
I'm not sure it reduces nesting since it would then become:
```python
        if as_backend_objs:
            return parameter
        elif isinstance(parameter, xr.DataArray):
            param_as_vals = self.apply_func(self._from_pyomo_param, parameter)
            if parameter.original_dtype.kind == "M":  # i.e., np.datetime64
                return xr.apply_ufunc(pd.to_datetime, param_as_vals)
            else:
                return param_as_vals.astype(parameter.original_dtype)
        else:
            return parameter
```

Or am I missing something?

Reply:
@jnnr: bringing your attention to this discussion as it is the blocking issue of this PR I think.

Reply:
Thanks for reminding me. The following should be equivalent to the existing code (the negation of the and turns it into an or).

```python 
if not isinstance(parameter, xr.DataArray) or as_backend_objs):
    return parameter

param_as_vals = self.apply_func(self._from_pyomo_param, parameter)
if parameter.original_dtype.kind == "M":  # i.e., np.datetime64
    return xr.apply_ufunc(pd.to_datetime, param_as_vals)
else:
    return param_as_vals.astype(parameter.original_dtype)
```